app/models/host/base.rb

```diff
@@ -246,7 +246,11 @@ def overwrite=(value)
     end
 
     def primary_interface
-      get_interface_by_flag(:primary)
+      # Returns a dummy object if host is unmanaged and without interfaces.
+      # Unmanaged hosts still have the interface requirement due to delegations and certain validations.
+      # A refactoring that allows unmanaged hosts to exist without interfaces would be welcome.
+      return get_interface_by_flag(:primary) if managed? || interfaces.detect(&:primary)
+      self.interfaces.build(:primary => true, :provision => true, :managed => false, :type => 'Nic::Managed')
```

Comment:
There's a method `build_required_interfaces` https://github.com/theforeman/foreman/blob/develop/app/models/host/base.rb#L325 that does the same as the line above plus it checks existence of provision interface. Could you use it instead?